packages/api/src/module/read/course-progress/course-progress.read-module.ts

```diff
@@ -25,4 +27,35 @@ export class CourseProgressReadModule {
       },
     });
   }
+
+  @OnEvent('LearningMaterialsTasks.TaskWasUncompleted')
+  async onTaskWasUncompleted(event: ApplicationEvent<TaskWasUncompleted>) {
+    const where = { learningMaterialsId: event.data.learningMaterialsId };
+    const courseProgress = await this.prismaService.courseProgress.findUnique({ where });
+
+    if (!courseProgress || courseProgress.learningMaterialsCompletedTasks === 0) {
+      return;
+    }
+
+    await this.prismaService.courseProgress.update({
```

Comment:
Whole method should be in transaction, but Prisma has better solution. Look how @KrystianKjjk has done it with increment. Do we have something silimar for decrement? Probably yes. 

In current implementation, when findUnique and update is asynchronous and not in the same transaction we may have situation that at line 34 we have completedTasks, let's say 5, but in line 43 it may be not true anymore (councurrent changes).

Reply:
I think this check `if (!courseProgress || courseProgress.learningMaterialsCompletedTasks === 0)` is still unnecessary, we can trust TaskWasUncompleted (is only for previously Completed task), because it's our internal event.
And the check may also return false positives, because it's async - as I've stated in previous comment.